src/__tests__/getJavaHome.test.ts

```diff
@@ -0,0 +1,119 @@
+import { getJavaHome } from "../getJavaHome";
+import { IJavaHomeInfo } from "locate-java-home/js/es5/lib/interfaces";
+
+describe("getJavaHome", () => {
+  const originalEnv = process.env;
+
+  afterEach(() => {
+    process.env = originalEnv;
+  });
+
+  it("reads from configuration", async () => {
+    const javaHomeConfig = "/path/to/java";
+    const javaHome = await getJavaHome(javaHomeConfig);
+    expect(javaHome).toBe(javaHomeConfig);
+  });
+
+  it("reads from JAVA_HOME", async () => {
+    const JAVA_HOME = "/path/to/java";
+    process.env = { ...originalEnv, JAVA_HOME };
+    const javaHome = await getJavaHome(undefined);
+    expect(javaHome).toBe(JAVA_HOME);
+  });
+
+  it("prefers configuration to JAVA_HOME and installed Java", async () => {
+    const javaHomeConfig = "/path/to/config/java";
+    const JAVA_HOME = "/path/to/java";
+    process.env = { ...originalEnv, JAVA_HOME };
+    mockLocateJavaHome([java8Jdk, java11Jdk]);
+    const javaHome = await require("../getJavaHome").getJavaHome(
+      javaHomeConfig
+    );
+    expect(javaHome).toBe(javaHomeConfig);
+  });
+
+  // NOTE(gabro): we don't care about testing locate-java-home since it's an external dependency
+  // and we assume it works as expected. However, we want to test how we select a specific version
+  // when multiple installed Java are available.
+
+  it("falls back to installed Java", async () => {
+    mockLocateJavaHome([java8Jdk, java11Jdk]);
+    const javaHome = await require("../getJavaHome").getJavaHome(undefined);
```

Comment:
we use `require` (instead of a static `import`) since we want this to be dynamically looked up and to pick up the latest mocked `locate-java-home`. See also `resetModules` below

Reply:
So I'm a bit confused by this. I just pulled this down, and locally all these are failing for me because this isn't picking up the mocked versions, but rather the values that my local `$JAVA_HOME` is set to?